tests/phpunit/unit/api/compiler/beans-page-compiler/compilePageStyles.php

```diff
@@ -0,0 +1,406 @@
+
+/**
+ * Class Tests_BeansPageCompiler_CompilePageStyles
+ *
+ * @package Beans\Framework\Tests\Unit\API\Compiler
+ * @group   api
+ * @group   api-compiler
+ */
+class Tests_BeansPageCompiler_CompilePageStyles extends Page_Compiler_Test_Case {
+
+	/**
+	 * Test _Beans_Page_Compiler::compile_page_styles() should not compile when the styles compiler is not supported.
+	 */
+	public function test_should_not_compile_when_styles_compiler_not_supported() {
+		Monkey\Functions\expect( 'beans_get_component_support' )
+			->once()
+			->with( 'wp_styles_compiler' )
+			->andReturn( false );
+		Monkey\Functions\expect( 'get_option' )->never();
+		Monkey\Functions\expect( '_beans_is_compiler_dev_mode' )->never();
+		Monkey\Functions\expect( 'beans_get' )->never();
+		Monkey\Functions\expect( 'beans_compile_css_fragments' )->never();
+
+		// Run the tests.
+		$this->assertNull( ( new _Beans_Page_Compiler() )->compile_page_styles() );
+	}
+
+	/**
+	 * Test _Beans_Page_Compiler::compile_page_styles() should not compile when the "compile all styles" option is not
+	 * set.
+	 */
+	public function test_should_not_compile_when_option_is_not_set() {
+		Monkey\Functions\when( 'beans_get_component_support' )->justReturn( true );
+		Monkey\Functions\expect( 'get_option' )
+			->once()
+			->with( 'beans_compile_all_styles', false )
+			->andReturn( false );
+		Monkey\Functions\expect( '_beans_is_compiler_dev_mode' )->never();
+		Monkey\Functions\expect( 'beans_get' )->never();
+		Monkey\Functions\expect( 'beans_compile_css_fragments' )->never();
+
+		// Run the tests.
+		$this->assertNull( ( new _Beans_Page_Compiler() )->compile_page_styles() );
+	}
+
+	/**
+	 * Test _Beans_Page_Compiler::compile_page_styles() should not compile when in dev mode.
+	 */
+	public function test_should_not_compile_when_in_dev_mode() {
+		Monkey\Functions\when( 'beans_get_component_support' )->justReturn( true );
+		Monkey\Functions\when( 'get_option' )->justReturn( true );
+		Monkey\Functions\expect( '_beans_is_compiler_dev_mode' )->once()->andReturn( true );
+		Monkey\Functions\expect( 'beans_get' )->never();
+		Monkey\Functions\expect( 'beans_compile_css_fragments' )->never();
+
+		// Run the tests.
+		$this->assertNull( ( new _Beans_Page_Compiler() )->compile_page_styles() );
+	}
+
+	/**
+	 * Test _Beans_Page_Compiler::compile_page_styles() should not compile when there are not styles.
+	 */
+	public function test_should_not_compile_when_there_are_not_styles() {
+		Monkey\Functions\when( 'beans_get_component_support' )->justReturn( true );
+		Monkey\Functions\when( 'get_option' )->justReturn( true );
+		Monkey\Functions\when( '_beans_is_compiler_dev_mode' )->justReturn( false );
+		Monkey\Functions\expect( 'add_query_arg' )->never();
+
+		// Check that beans_get() only gets called once.
+		Monkey\Functions\expect( 'beans_get' )->once()->andReturn( null );
+
+		// Check that this function does not get called.
```

Comment:
For consistency with the other tests:
Check that beans_compile_css_fragments() does not get called.